neo/tools/compilers/fontcodec/blFontCodec.cpp

```diff
@@ -0,0 +1,199 @@
+				  --remap, --remapdir=VALUE	directory in which The Dark Mod's remap tables are (like: polish.map)
+*/
+
+	common->Printf(
+		"Usage: fontCodec [options] filename\n"
+		"Options:\n"
+		"-h, --help						Show this help message and exit.\n"
+		"-d, --decompile				Decompile instead of compiling\n"
+		"-v								Verbose\n"
+		"\n in case of compilations the input folder will be 'fonts', in decompilations 'newfonts'\n"
+		//"	 --output=VALUE			File name of the output name different than the default '48.dat' name."
+	);
+}
+
+/*
+============
+blFontCodec::GatherComandArgs
+============
+*/
+
+idStr blFontCodec::GatherComandArgs( const idCmdArgs& args ) {
+	int			i;
+
+	for( i = 1 ; i < args.Argc() ; i++ )
+	{
+		const char* s;
+
+		s = args.Argv( i );
+		if( s[0] == '-' )
+		{
+			s++;
+			if( s[0] == '\0' )
+			{
+				continue;
+			}
+		}
+
+		if( !idStr::Icmp( s, "h" ) )
+		{
+			FontCompHelp();
+			return '\0';
+		}
+		else if( !idStr::Icmp( s, "help" ) )
+		{
+			FontCompHelp();
+			return '\0';
+		}
+		else if( !idStr::Icmp( s, "d" ) )
+		{
+			fontCodecGlobals.decompile = true;
+		}
+		else if( !idStr::Icmp( s, "decompile" ) )
+		{
+			fontCodecGlobals.decompile = true;
+		}
+		else if( !idStr::Icmp( s, "v" ) )
+		{
+			common->Printf( "verbose = true\n" );
+			fontCodecGlobals.verbose = true;
+		}
+		/*
+		else if( !idStr::Icmp( s, "output" ) )
+		{
+			common->Printf( "drawflag = true\n" );
+			dmapGlobals.drawflag = true;
+		}
+		*/
+		else
+		{
+			break;
+		}
+	}
+
+	if( i >= args.Argc() )
+	{
+		common->Error( "Usage: fontCodec [options] filename\n" );
+	}
+
+	return args.Argv( i );
+}
+
+/*
+============;
+blFontCodec::fontCodec
+============
+*/
+
+void blFontCodec::FontCodec( const idCmdArgs& args ){
+	int			i;
+	idStr		q_path_to_file;
+	idStr		filename;
+
+	if( args.Argc() < 2 ) {
+		FontCompHelp();
+		return;
+	}
+
+	q_path_to_file = blFontCodec::GatherComandArgs( args );		// may have an extension
+	if( q_path_to_file.c_str() == '\0' ) {
+					return;
+	}
+
+	if( fontCodecGlobals.decompile ) {
+		fontCodecGlobals.inputDirectory = "newfonts";
+		i = 8; // 'newfonts/' --> 8 characters
+	} else {
+		fontCodecGlobals.inputDirectory = "fonts";
+		i = 5;
+	}
+
+	q_path_to_file.BackSlashesToSlashes();
+
+	if( q_path_to_file.Icmpn( fontCodecGlobals.inputDirectory + "/", i ) != 0 )
+	{
+		filename = q_path_to_file;
+		q_path_to_file = fontCodecGlobals.inputDirectory + "/" + q_path_to_file;
+	}
+
+	q_path_to_file.StripFileExtension();
+	filename.StripFileExtension();
+
+	if( fontCodecGlobals.decompile ) {
+		filename += ".fnt"
+		fontCodecGlobals.outputFilename = filename; // we only do this in decompile because the default accounts for the compile.
+		q_path_to_file += ".dat"
+		common->Printf( "---- decompiling font ----\n" );
+	} else {
+		q_path_to_file += ".fnt"
+		common->Printf( "---- compiling font ----\n" );
+	}
+	fontCodecGlobals.inputFilename = q_path_to_file;
+
+	BM_font = new BMfont( this );
```

Comment:
BMfont doesnt have a constructor supporting a call like that, remove the this should do the trick